components/script/dom/htmlbuttonelement.rs

```diff
@@ -77,6 +77,11 @@ impl HTMLButtonElement {
             document,
         )
     }
+
+    #[inline]
+    pub fn button_type(&self) -> ButtonType {
```

Comment:
I would rather have a `is_submit` method on HTMLButtonElement, since the `button_type` seems to only be used for that purpose(thay way the `ButtonType` doesn't have to become public either).

Reply:
Sounds reasonable, done.